pkg/domainservice/adapterInterfaces.go

```diff
@@ -79,17 +80,17 @@ type RemoteDoguRegistry interface {
 	// GetDogu returns the dogu specification for the given dogu and version or
 	// an NotFoundError indicating that there was no dogu spec found or
 	// an InternalError indicating that the caller has no fault.
-	GetDogu(qualifiedDoguName string, version string) (*core.Dogu, error)
+	GetDogu(doguName common.QualifiedDoguName, version string) (*core.Dogu, error)
 
 	// GetDogus returns the all requested dogu specifications or
 	// an NotFoundError indicating that any dogu spec was not found or
 	// an InternalError indicating that the caller has no fault.
-	GetDogus(dogusToLoad []DoguToLoad) (map[string]*core.Dogu, error)
+	GetDogus(dogusToLoad []DoguToLoad) (map[common.QualifiedDoguName]*core.Dogu, error)
 }
 
 type DoguToLoad struct {
-	QualifiedDoguName string
-	Version           string
+	DoguName common.QualifiedDoguName
```

Comment:
```suggestion
	QualifiedDoguName common.QualifiedDoguName
```
Maybe we should stick to "QualifiedDoguName" so "DoguName" won't get mistaken with QualifiedDoguName.Name